core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -424,6 +424,7 @@ public Location add(Operation operation) throws IOException {
             closeOnTragicEvent(ex);
             throw ex;
         } catch (Throwable e) {
```

Comment:
I think you misunderstood the logic here. We only close ourself iff the current writer has a tragic event. It's unrelated to the actual exception we are catching here. The Exception is only passed to the `closeOnTragicEvent` call to add a suppressed exception from the close call if we do it. But this is only the trigger to check if we hit a tragic event and I think we should do this all the time if something goes wrong.